src/Symfony/Component/PropertyInfo/PropertyInfoCacheExtractor.php

```diff
@@ -106,7 +106,7 @@ private function extract($method, array $arguments)
             return call_user_func_array(array($this->propertyInfoExtractor, $method), $arguments);
         }
 
-        $key = $this->escape($method.'.'.$serializedArguments);
+        $key = rawurlencode($method.'.'.$serializedArguments);
```

Comment:
Can you add a comment above to explain why we need to encode (that's not trivial to understand now that the comment on the escape method is not here anymore)